### PR TITLE
Adding Windows support for system agent installer.

### PR DESCRIPTION
**Pull request by phillipsj**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -8,13 +8,15 @@ platform:
   arch: amd64
 
 steps:
+
```
Comment (luthermonson) [inline .drone.yml:11]: nit

File: .drone.yml [hunk 2]
```diff
@@ -8,13 +8,15 @@ platform:
...
 - name: download-linux-amd64
   image: registry.suse.com/suse/sle15:15.3.17.8.1
   environment:
     ARCH: amd64
   commands:
     - zypper -n install curl jq git
     - scripts/download
+
```
Comment (luthermonson) [inline .drone.yml:19]: nit

**Comment by dereknola** [comment]
Can we get an issue attached to this PR.

**Comment by phillipsj** [comment]
@dereknola none of the previous PRs have had issues attached and there appears to be only one existing issue. Is this going to be a new process? If so, what is it going to be look like? @cwayne18 @sirredbeard

**Comment by dereknola** [comment]
@phillipsj Nevermind, you are correct most of the PRs do not have issues attached. I reacted too quickly when I saw this PR and assumed it was in the rke2 repo, which do need issues attached.

File: scripts/windows/version.ps1 [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+$TREE_STATE = "clean"
+$COMMIT = $env:DRONE_COMMIT
+$TAG = $env:DRONE_TAG
+
+if (-not $TAG) {
+    if (Test-Path -Path $env:DAPPER_SOURCE\.git) {
+        Push-Location $env:DAPPER_SOURCE
+        if ("$(git status --porcelain --untracked-files=no)") {
+            $env:DIRTY = "dirty"
+            TREE_STATE = "dirty"
```
Comment (luthermonson) [inline scripts/windows/version.ps1:20]: dolla holla

File: scripts/windows/version.ps1 [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    $REPO = "rancher"
+}
+else {
+    $env:REPO = "rancher"
+}
+
+#export stuff out 
+$env:VERSION = $VERSION
+$env:COMMIT = $COMMIT
+$env:REPO = $REPO
```
Comment (luthermonson) [inline scripts/windows/version.ps1:90]: so always "rancher" then?

File: scripts/windows/version.ps1 [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    }
+}
+else {
+    # validate the tag format and create our VERSION variable
+    if (-not ($TAG -match '^v[0-9]{1}\.[0-9]{2}\.[0-9]+-*[a-zA-Z0-9]*\+rke2r[0-9]+$')) {
+        Write-Host "Tag does not match our expected format. Exiting."
+        exit 1
+    }
+
+    $VERSION=$TAG
```
Comment (luthermonson) [inline scripts/windows/version.ps1:74]: nit

**Review by luthermonson — APPROVED** [review]

**Review by dereknola — APPROVED** [review]

**Review by dweomer — APPROVED** [review]

**Pull request merged**